src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -0,0 +1,253 @@
+        target_coords,
+    const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  DataVector solutions{target_coords[0]};
+  for (auto& solution : solutions) {
+    solution = inverse<double>({{solution}}, time, map_list)[0];
+  }
+  return {{solutions}};
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::frame_velocity(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  const auto frame_vel =
+      source_coords[0] * (map_list.at(f_of_t_a_).func_and_deriv(time)[1][0] +
+                          (map_list.at(f_of_t_b_).func_and_deriv(time)[1][0] -
+                           map_list.at(f_of_t_a_).func_and_deriv(time)[1][0]) *
+                              square(source_coords[0] / outer_boundary_));
+
+  return {{frame_vel}};
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> CubicScale::jacobian(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> jac{
+      make_with_value<tt::remove_cvref_wrap_t<T>>(
+          dereference_wrapper(source_coords[0]), 0.0)};
+
+  get<0, 0>(jac) = map_list.at(f_of_t_a_).func(time)[0][0] +
+                   3.0 * (map_list.at(f_of_t_b_).func(time)[0][0] -
+                          map_list.at(f_of_t_a_).func(time)[0][0]) *
+                       square(source_coords[0] / outer_boundary_);
+
+  return jac;
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame>
+CubicScale::inv_jacobian(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> inv_jac{
+      make_with_value<tt::remove_cvref_wrap_t<T>>(
+          dereference_wrapper(source_coords[0]), 0.0)};
+
+  get<0, 0>(inv_jac) = 1.0 / (map_list.at(f_of_t_a_).func(time)[0][0] +
+                              3.0 * (map_list.at(f_of_t_b_).func(time)[0][0] -
+                                     map_list.at(f_of_t_a_).func(time)[0][0]) *
+                                  square(source_coords[0] / outer_boundary_));
+
+  return inv_jac;
+}
+
+template <typename T>
+tnsr::Iaa<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> CubicScale::hessian(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  auto result{
+      make_with_value<tnsr::Iaa<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame>>(
+          dereference_wrapper(source_coords[0]), 0.0)};
+  // time-time
+  get<0, 0, 0>(result) =
+      map_list.at(f_of_t_a_).func_and_2_derivs(time)[2][0] * source_coords[0] +
```

Comment:
if `func_and_...` does computations on f_of_a_, b, it would be good to factor them out and only do the computation once